pyanaconda/modules/payloads/payload/dnf/dnf.py

```diff
@@ -358,18 +382,98 @@ def get_repo_configurations(self):
         """
         return list(filter(None, [s.generate_repo_configuration() for s in self.sources]))
 
+    def set_up_sources_with_task(self):
+        """Set up installation sources."""
+        task = SetUpDNFSourcesTask(
+            sources=self.sources,
+            repositories=self.repositories,
+            configuration=self.packages_configuration,
+        )
+        task.succeeded_signal.connect(
+            lambda: self._set_up_sources_on_success(task.get_result())
+        )
+        return task
+
+    def _set_up_sources_on_success(self, result: SetUpDNFSourcesResult):
+        """Update the module based on the configured sources."""
+        self._set_dnf_manager(result.dnf_manager)
+        self.set_repositories(result.repositories)
+        self._internal_sources += result.sources
+
+    def tear_down_sources_with_task(self):
+        """Tear down installation sources."""
+        return TearDownDNFSourcesTask(
+            dnf_manager=self.dnf_manager,
+            sources=self._internal_sources + self.sources
+        )
+
     def install_with_tasks(self):
         """Install the payload.
 
         :return: list of tasks
         """
-        # TODO: Implement this method
-        return []
+        tasks = [
+            SetRPMMacrosTask(
+                configuration=self.packages_configuration
+            ),
+            ResolvePackagesTask(
+                dnf_manager=self.dnf_manager,
+                selection=self.packages_selection,
+            ),
+            PrepareDownloadLocationTask(
+                dnf_manager=self.dnf_manager,
+            ),
+            DownloadPackagesTask(
+                dnf_manager=self.dnf_manager,
+            ),
+            InstallPackagesTask(
+                dnf_manager=self.dnf_manager,
+            ),
+            CleanUpDownloadLocationTask(
+                dnf_manager=self.dnf_manager,
+            ),
+        ]
+
+        self._collect_kernels_on_success(InstallPackagesTask, tasks)
```

Comment:
This is kind of a common issue, so it would be great to discuss possible options and unify this across all modules. I will schedule a meeting for that.